src/components/Content.tsx

```diff
@@ -0,0 +1,67 @@
+import { useEffect, useState } from "react";
+import axios from "axios";
+import Swal from 'sweetalert2'
+import AllToDos from "./AllToDos";
+import Form from "./Form";
+
+function Content(){
+
+    const [data, setData] = useState({
+        todos: [] as {id: number, title: string; completed: boolean }[],
+        isLoading: false,
+        error: ""
+    })
+
+    useEffect(()=>{
+        const fetchData= async ()=>{
+            try{
+                const todosList= (await axios.get("https://jsonplaceholder.typicode.com/todos")).data;
+                setData({...data, todos: todosList.slice(0, 10), isLoading: true}) // take just 10 elements instead of all the return from the API (200 elements)
+            }catch(e:any){
+                setData({...data, error:e, isLoading: true})
+            }
+        }
+        fetchData();
+    },[])
+
+    async function onSubmitSuccess(newToDo:{title: string, completed: boolean}){
+        const element= await axios.post("https://jsonplaceholder.typicode.com/todos", newToDo)
+         setData({...data, todos:[...data.todos, element.data]})
+    }
+
+    function handleCheck(id: number){
+        const updatedTodos = data.todos.map(ele =>{
+            if(ele.id == id)
+                return {...ele, completed: !ele.completed}
+            return ele
+        })
+        setData({...data, todos: updatedTodos})
+    }
+
+    function showDeletePopup(id: number){
+        Swal.fire({
+            text: "Are You sure You want delete this todo ? ",
+            title: "Cation !",
+            icon: "warning",
+            confirmButtonColor: "#FFA07A",
+            showDenyButton: true
+        }).then(ret => {
+            if(ret.isConfirmed)
+            handleDelete(id)
+        })
+    }
+
+    async function handleDelete(id: number){
+        await axios.delete(`https://jsonplaceholder.typicode.com/todos/${id}`);
+        const updatedTodos= data.todos.filter((ele)=> ele.id !== id);
+        setData({...data, todos: updatedTodos})
```

Comment:
you should wrap it in try-catch, in case the deletion fails you should give an indication to the user.

and instead of filtering the todos, usually we should get the data from the BE again